staging/src/k8s.io/component-base/metrics/registry_test.go

```diff
@@ -317,3 +314,80 @@ func TestValidateShowHiddenMetricsVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestEnableHiddenMetrics(t *testing.T) {
+	currentVersion := apimachineryversion.Info{
+		Major:      "1",
+		Minor:      "17",
+		GitVersion: "v1.17.1-alpha-1.12345",
+	}
+
+	var tests = []struct {
+		name           string
+		fqName         string
+		counter        *Counter
+		mustRegister   bool
+		expectedMetric string
+	}{
+		{
+			name:   "hide by register",
+			fqName: "hidden_metric_register",
+			counter: NewCounter(&CounterOpts{
+				Name:              "hidden_metric_register",
+				Help:              "counter help",
+				StabilityLevel:    STABLE,
+				DeprecatedVersion: "1.16.0",
+			}),
+			mustRegister: false,
+			expectedMetric: `
+				# HELP hidden_metric_register [STABLE] (Deprecated since 1.16.0) counter help
+				# TYPE hidden_metric_register counter
+				hidden_metric_register 1
+				`,
+		},
+		{
+			name:   "hide by must register",
+			fqName: "hidden_metric_must_register",
+			counter: NewCounter(&CounterOpts{
+				Name:              "hidden_metric_must_register",
+				Help:              "counter help",
+				StabilityLevel:    STABLE,
+				DeprecatedVersion: "1.16.0",
+			}),
+			mustRegister: true,
+			expectedMetric: `
+				# HELP hidden_metric_must_register [STABLE] (Deprecated since 1.16.0) counter help
+				# TYPE hidden_metric_must_register counter
+				hidden_metric_must_register 1
+				`,
+		},
+	}
+
+	for _, test := range tests {
+		tc := test
+		t.Run(tc.name, func(t *testing.T) {
+			registry := newKubeRegistry(currentVersion)
+			if tc.mustRegister {
+				registry.MustRegister(tc.counter)
+			} else {
+				_ = registry.Register(tc.counter)
+			}
+
+			tc.counter.Inc() // no-ops, because counter hasn't been initialized
+			if err := testutil.GatherAndCompare(registry, strings.NewReader(""), tc.fqName); err != nil {
+				t.Fatal(err)
```

Comment:
nit: let's pass in a message here along with the actual error

Reply:
Actually, don't need to since the output is clear enough.

e.g. 
```
    --- FAIL: TestEnableHiddenMetrics/hide_by_must_register (0.00s)
        registry_test.go:389: 
            metric output does not match expectation; want:
            
            # HELP hidden_metric_must_register [STABLE] (Deprecated since 1.16.0) counter help
            # TYPE hidden_metric_must_register counter
            hidden_metric_must_register 2
            
            got:
            
            # HELP hidden_metric_must_register [STABLE] (Deprecated since 1.16.0) counter help
            # TYPE hidden_metric_must_register counter
            hidden_metric_must_register 1
```